### PR TITLE
Add base networking templates for Azure

### PR DESCRIPTION
**Pull request by scottslowe**
This PR adds base networking templates for Azure in Go, Python, TypeScript, and YAML. Templates in C# and Java will be added later via separate PRs.

This PR replaces #379.

### CODE REVIEW

File: base-networking-azure-go/main.go [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+					pulumi.String("10.0.0.0/16"),
+				},
+			},
+			ResourceGroupName: resourceGroup.Name,
+		})
+		if err != nil {
+			return err
+		}
+
+		// Create three subnets in the virtual network
```
Comment (joeduffy) [inline base-networking-azure-go/main.go:30]: Any way to use a for loop for this?

Comment (scottslowe) [reply base-networking-azure-go/main.go:30]: Yeah, I can rewrite the Go/Python/TypeScript templates to use a loop. Good call.

Comment (scottslowe) [reply base-networking-azure-go/main.go:32]: Addressed via bd2863c; the Go, Python, and TypeScript templates now use loops to create the subnets.

**Comment by cnunciato** [comment]
Hey @scottslowe, do we want to keep these base-networking PRs open, or close until we come back to them?

**Comment by scottslowe** [comment]
Let's close the PRs until we are ready to actually work on these templates. Thanks!

**Comment by scottslowe** [comment]
Closing until we are ready to proceed with working on the base networking templates

**Pull request closed without merging**